azafea/event_processors/endless/metrics/tests/integration/test_metrics_cli.py

```diff
@@ -1058,3 +1058,33 @@ def test_remove_os_info_no_quotes(self, capfd):
 
         capture = capfd.readouterr()
         assert 'No OS info with extra quotes in database' in capture.out
+
+    def test_refresh_views(self):
+        from azafea.event_processors.endless.metrics.request import Request, RequestByDay
+
+        # Create the table
+        self.run_subcommand('initdb')
+        self.ensure_tables(Request)
+
+        occured_at = datetime.utcnow().replace(tzinfo=timezone.utc)
+
+        with self.db as dbsession:
+            dbsession.add(Request(serialized=b'whatever', sha512='sha512-1', received_at=occured_at,
+                                  absolute_timestamp=1, relative_timestamp=2, machine_id='machine1',
+                                  send_number=0))
+            dbsession.add(Request(serialized=b'whatever', sha512='sha512-2', received_at=occured_at,
+                                  absolute_timestamp=3, relative_timestamp=4, machine_id='machine2',
+                                  send_number=0))
+            dbsession.add(Request(serialized=b'whatever', sha512='sha512-3', received_at=occured_at,
+                                  absolute_timestamp=5, relative_timestamp=6, machine_id='machine2',
+                                  send_number=0))
+
+        with self.db as dbsession:
+            assert dbsession.query(RequestByDay).count() == 0
+
+        self.run_subcommand('refresh-views')
+
+        with self.db as dbsession:
+            req1, req2 = dbsession.query(RequestByDay).order_by(RequestByDay.machine_id).all()
+            assert (req1.machine_id, req1.day) == ('machine1', occured_at.date())
+            assert (req2.machine_id, req2.day) == ('machine2', occured_at.date())
```

Comment:
I think this answers my question above about `DISTINCT` versus `DISTINCT ON`, but I think this test would be more meaningful if the request table contained rows with different dates (as opposed to three events with identical timestamps).

Reply:
> I think this answers my question above about `DISTINCT` versus `DISTINCT ON`, but I think this test would be more meaningful if the request table contained rows with different dates (as opposed to three events with identical timestamps).

Tests can definitely be improved, as said in my comment.

Reply:
I’ve improved this test, but feel free to add other comments if you think that it’s not good enough.